src/entrypoints/recorder.ts

```diff
@@ -326,34 +342,50 @@ function initXhrObserver(cb: networkCallback, win: IWindow, options: Required<Ne
  *  NB PerformanceNavigationTiming extends PerformanceResourceTiming
  *  Here we don't care which interface it implements as both expose `serverTimings`
  */
-const exposesServerTiming = (event: PerformanceEntry): event is PerformanceResourceTiming =>
-    event.entryType === 'navigation' || event.entryType === 'resource'
-
-function prepareRequest(
-    entry: PerformanceResourceTiming,
-    method: string | undefined,
-    status: number | undefined,
-    networkRequest: Partial<CapturedNetworkRequest>,
+const exposesServerTiming = (event: PerformanceEntry | null): event is PerformanceResourceTiming =>
+    !!event && (event.entryType === 'navigation' || event.entryType === 'resource')
+
+function prepareRequest({
+    entry,
+    method,
+    status,
+    networkRequest,
+    isInitial,
+    start,
+    end,
+}: {
+    entry: PerformanceResourceTiming | null
+    method: string | undefined
+    status: number | undefined
+    networkRequest: Partial<CapturedNetworkRequest>
     isInitial?: boolean
-): CapturedNetworkRequest[] {
+    start?: number
+    end?: number
```

Comment:
Any reason to make these optional? Looks like this method is only called with the results of `window.performance.now()` as inputs for these values which always returns a number

Reply:
only that the values start undefined... _really_ you shouldn't ever get to the point of needing them and the value not be present but 🙈